bin_tests/tests/crashtracker_bin_test.rs

```diff
@@ -0,0 +1,53 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+#![cfg(unix)]
+
+use std::fs;
+use std::process;
+
+use bin_tests::{build_artifacts, ArtifactType, ArtifactsBuild, Profile};
+
+#[test]
+#[cfg_attr(miri, ignore)]
+fn test_crash_tracking_bin() {
+    let crash_tracking_receiver = ArtifactsBuild {
+        name: "profiling-crashtracking-receiver".to_owned(),
+        profile: Profile::Debug,
+        artifact_type: ArtifactType::ExecutablePackage,
+        triple_target: None,
+    };
+    let crashtracker_bin = ArtifactsBuild {
+        name: "crashtracker_bin".to_owned(),
+        profile: Profile::Debug,
```

Comment:
Might be interesting to do a prop-test covering both debug and release builds?

Reply:
Added.
Interestingly, the null pointer dereference gets optimized out in release mod, and we receive a SIGTRAP instead.  I guess it's because it's an undefined behavior